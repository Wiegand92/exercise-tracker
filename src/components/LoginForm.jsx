import React, {useState} from 'react';

const LoginForm = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const handleCreate = e => {
    e.preventDefault();
    const user = {username, password};

    fetch('/users/add', {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify(user),
    })
      .then(() => {
        setUsername('');
        setPassword('');
        alert('User Created!');
      })
      .catch(err => alert(`Error: ${err}`));
  };

  const handleLogin = e => {
    e.preventDefault();
    const user = {username, password};

    fetch('/users/login', {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify(user),
    })
      .then(res => res.json())
      .then(data => {
        console.log(data);
        if (data.user) {
          localStorage.setItem('token', data.auth);
          window.location = '/';
        }
      })
      .catch(err => console.error(err));
  };

  return (
    <form className="login-form" onSubmit={handleLogin}>
      <h2>Login</h2>
      Username:{' '}
      <input
        type="text"
        value={username}
        onChange={e => setUsername(e.target.value)}
        name="username"
        autoComplete="off"
        required
      />
      Password:{' '}
      <input
        type="password"
        value={password}
        onChange={e => setPassword(e.target.value)}
        name="password"
        required
      />
      <input type="submit" value="Login" />
      <button onClick={handleCreate}>Create User</button>
    </form>
  );
};

export default LoginForm;
