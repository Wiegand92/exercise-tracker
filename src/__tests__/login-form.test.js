import React from 'react';
import {render, screen} from '@testing-library/react';
import LoginForm from '../components/LoginForm';

test('The login form renders', () => {
  render(<LoginForm />);
  const form = screen.getByRole('heading');
  expect(form).not.toBeUndefined();
});
