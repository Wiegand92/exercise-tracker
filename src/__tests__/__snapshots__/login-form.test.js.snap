// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The login form renders with all its components 1`] = `
<DocumentFragment>
  <form
    class="login-form"
  >
    <h2>
      Login
    </h2>
    <label
      for="username"
    >
      Username:
    </label>
    <input
      autocomplete="off"
      id="username"
      required=""
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password:
    </label>
    <input
      id="password"
      required=""
      type="password"
      value=""
    />
    <input
      type="submit"
      value="Login"
    />
    <button>
      Create User
    </button>
  </form>
</DocumentFragment>
`;
